src/interpolate.jl

```diff
@@ -40,6 +40,113 @@ function cell2node(cell_centered_data::AbstractArray{Float64})
 end
 
 
+# Convert 3d unstructured data to 2d slice.
+# Additional to the new unstructured data updated coordinates, levels and
+# center coordinates are returned.
+function unstructured_2d_to_3d(unstructured_data::AbstractArray{Float64},
+                               coordinates::AbstractArray{Float64},
+                               levels::AbstractArray{Int}, length_level_0::Float64,
+                               center_level_0::AbstractArray{Float64},
+                               slice_axis, slice_axis_intersect)
+
+  dimensions = Dict(
+    :x => 1,
+    :y => 2,
+    :z => 3
+  )
+  if !haskey(dimensions, slice_axis)
+    supported_dims = keys(dimensions)
+    error("illegal dimension '$slice_axis', supported dimensions are $supported_dims")
+  end
+  slice_axis_dimension = dimensions[slice_axis]
+  other_dimensions = [1, 2, 3][1:end .!= slice_axis_dimension]
+
+  # Extract data shape information
+  n_nodes_in, _, _, n_elements, n_variables = size(unstructured_data)
+
+  # Get node coordinates for DG locations on reference element
+  nodes_in, _ = gauss_lobatto_nodes_weights(n_nodes_in)
+
+  # New unstructured data has one dimension less.
+  # The redundant element ids are removed later.
+  new_unstructured_data = similar(unstructured_data[1, ..])
```

Comment:
I think you need a `@views` here, otherwise the slice operation will copy the data:
```suggestion
  @views new_unstructured_data = similar(unstructured_data[1, ..])
```

Reply:
So `similar` would point to some unallocated space in memory and in my version without `@views` I would copy the (random) data from this unallocated memory space somewhere else in memory only to overwrite it later. Did I understand it correctly?

Reply:
`similar(unstructured_data[1, ..])` allocates twice: At first, `unstructured_data[1, ..]` is creating a new `Array`. Then, `similar` allocates an additional array. With `@views`, you replace `unstructured_data[1, ..]` by `view(unstructured_data, 1, ..)`, which doesn't allocate (on Julia v1.5 and newer).

Reply:
Ah, I see. Thank you!